File: ResQNet_Project/resqnet-backend/src/main/java/com/pu_deltaforce/resqnet_backend/model/CivicReport.java
package com.pu_deltaforce.resqnet_backend.model;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

@Entity
@Table(name = "civic_reports")
@Data
public class CivicReport {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String reporterName;

    @Column(nullable = false)
    private String reporterPhone;

    @Column(nullable = false)
    private String location;

    private Double latitude;
    private Double longitude;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ReportType reportType;

    @Column(length = 1000)
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ReportStatus status;

    @Column(nullable = false)
    private LocalDateTime reportedAt;

    public enum ReportType {
        ROAD_DAMAGE, WATERLOGGING, POWER_OUTAGE,
        GAS_LEAK, FIRE, OTHER
    }

    public enum ReportStatus {
        PENDING, IN_PROGRESS, RESOLVED
    }

    @PrePersist
    public void prePersist() {
        this.status = ReportStatus.PENDING;
        this.reportedAt = LocalDateTime.now();
    }
}

File: ResQNet_Project/resqnet-frontend/src/app/core/services/civic-report.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { environment } from '../../../environments/environment';

export interface CivicReport {
  id?: number;
  title: string;
  description: string;
  reportType: 'ROAD_DAMAGE' | 'WATERLOGGING' | 'POWER_OUTAGE' | 'GAS_LEAK' | 'FIRE' | 'OTHER';
  status?: 'PENDING' | 'IN_PROGRESS' | 'RESOLVED';
  location: string;
  reporterName: string;
  reporterPhone: string;
  createdAt?: string;
}

@Injectable({ providedIn: 'root' })
export class CivicReportService {
  private apiUrl = `${environment.apiUrl}/civic-reports`;

  constructor(private http: HttpClient) { }

  getAll() { return this.http.get<CivicReport[]>(this.apiUrl); }
  create(data: CivicReport) { return this.http.post<CivicReport>(this.apiUrl, data); }
  updateStatus(id: number, status: string) {
    return this.http.put(`${this.apiUrl}/${id}/status?status=${status}`, {});
  }
  getByStatus(status: string) {
    return this.http.get<CivicReport[]>(`${this.apiUrl}/status/${status}`);
  }
  getByType(type: string) {
    return this.http.get<CivicReport[]>(`${this.apiUrl}/type/${type}`);
  }
}